modules/core/src/v2/coins/utxo/recovery/crossChainRecovery.ts

```diff
@@ -41,607 +52,354 @@ class BitgoPublicApi {
-      if (input.chain === 10 || input.chain === 11) {
-        txInfo.nSegwitInputs++;
-      } else {
-        txInfo.nP2SHInputs++;
-      }
-    }
-
-    txInfo.nOutputs = 1;
-    txInfo.unspents = _.map(
-      this.txInfo.inputs,
-      _.partialRight(_.pick, ['chain', 'index', 'redeemScript', 'id', 'address', 'value'])
-    );
-    txInfo.changeAddresses = [];
-    txInfo.walletAddressDetails = {};
-
-    const feeInfo: any = {};
+function getTxInfo(
+  transaction: utxolib.bitgo.UtxoTransaction,
+  unspents: WalletUnspent[],
+  walletId: string,
+  walletKeys: RootWalletKeys
+): TransactionInfo {
+  const inputAmount = utxolib.bitgo.unspentSum(unspents);
+  const outputAmount = transaction.outs.reduce((sum, o) => sum + o.value, 0);
+  const outputs = transaction.outs.map((o) => ({
+    address: utxolib.address.fromOutputScript(o.script, transaction.network),
+    valueString: o.value.toString(),
+    change: false,
+  }));
+  const inputs = unspents.map((u) => {
+    // NOTE:
+    // The `redeemScript` and `walletScript` properties are required for legacy versions of BitGoJS
+    // which might require these scripts for signing. The Wallet Recovery Wizard (WRW) can create
+    // unsigned prebuilds that are submitted to BitGoJS instances which are not necessarily the same
+    // version.
+    const addressKeys = walletKeys.deriveForChainAndIndex(u.chain, u.index);
+    const scriptType = scriptTypeForChain(u.chain);
+    const { redeemScript, witnessScript } = outputScripts.createOutputScript2of3(addressKeys.publicKeys, scriptType);
 
-    feeInfo.size =
-      VirtualSizes.txOverheadSize +
-      VirtualSizes.txP2shInputSize * this.txInfo.inputs.length +
-      VirtualSizes.txP2pkhOutputSize;
+    return {
+      ...u,
+      wallet: walletId,
+      fromWallet: walletId,
+      redeemScript: redeemScript?.toString('hex'),
+      witnessScript: witnessScript?.toString('hex'),
+    } as WalletUnspentLegacy;
+  });
+  return {
+    inputAmount,
+    outputAmount,
+    minerFee: inputAmount - outputAmount,
+    spendAmount: outputAmount,
+    inputs,
+    unspents: inputs,
+    outputs,
+    externalOutputs: outputs,
+    changeOutputs: [],
+    payGoFee: 0,
+  } as TransactionInfo;
+}
 
-    feeInfo.feeRate = this.feeRates[this.sourceCoin.type];
-    feeInfo.fee = Math.round((feeInfo.size / 1000) * feeInfo.feeRate);
-    feeInfo.payGoFee = 0;
-    feeInfo.payGoFeeString = '0';
+function getFeeInfo(transaction: utxolib.bitgo.UtxoTransaction, unspents: WalletUnspent[]): FeeInfo {
+  const vsize = Dimensions.fromUnspents(unspents).plus(Dimensions.fromOutputs(transaction.outs)).getVSize();
+  const inputAmount = utxolib.bitgo.unspentSum(unspents);
+  const outputAmount = transaction.outs.reduce((sum, o) => sum + o.value, 0);
+  const fee = inputAmount - outputAmount;
+  return {
+    size: vsize,
+    fee,
+    feeRate: fee / vsize,
+    payGoFee: 0,
+  };
+}
 
+export async function recoverCrossChain(
```

Comment:
A comment block would be good here too explaining what this function does since it's exported.

Reply:
will do